src/telemetry/telemetry.c

```diff
@@ -89,29 +112,31 @@ telemetry_parse_version(const char *json, VersionInfo *installed_version, Versio
  * called "current_timescaledb_version". Check this against the local
  * version, and notify the user if it is behind.
  */
-static bool
+static void
 process_response(const char *json)
 {
-	VersionInfo installed_version;
 	VersionResult result;
+	bool		is_uptodate = DatumGetBool(DirectFunctionCall2(texteq,
+															   DirectFunctionCall2(json_object_field_text,
+																				   CStringGetTextDatum(json),
+																				   PointerGetDatum(cstring_to_text(TS_IS_UPTODATE_JSON_FIELD))),
+															   PointerGetDatum(cstring_to_text("true"))));
 
-	version_get_info(&installed_version);
-
-	if (!telemetry_parse_version(json, &installed_version, &result))
-	{
-		elog(WARNING, "could not get TimescaleDB version from server response: %s", result.errhint);
-		return false;
-	}
-
-	if (result.is_up_to_date)
+	if (is_uptodate)
 		elog(NOTICE, "the \"%s\" extension is up-to-date", EXTENSION_NAME);
 	else
+	{
+		if (!validate_server_version(json, &result))
+		{
+			elog(WARNING, "server did not return a valid TimescaleDB version: %s", result.errhint);
```

Comment:
do we want to say "version-check server did not return...."

Reply:
hm, think that's unnecessary and potentially confusing since that's not the only thing the server does.